scripts/cmd_args.py

```diff
@@ -64,191 +64,210 @@ def type_or_none_fn(type):
+                     help='list of data files in the form of "[file1, file2, '
+                          '...]";'
+                          'or a single path to a file or folder of files')
+    parser_train.add('--checkpoint_fname', required=True, type=str,
+                     help="checkpoint filename to save the model")
+    parser_train.add('--save_freq', required=True, type=int,
+                     help="model checkpoint saving frequency")
+    # Learning args
+    parser_train.add('--clip_grad', required=True, type=float,
+                     help='Grad clipping for bad/extreme batches')
+    parser_train.add('--lr', required=True, type=float,
+                     help='learning rate')
+    parser_train.add('--epochs', required=True, type=int,
+                     help='Number of epochs')
+    parser_train.add('--mse_weight', required=True, type=float,
+                     help='relative weight of mse loss')
+    parser_train.add('--pearson_weight', required=True, type=float,
+                     help='relative weight of pearson correlation loss')
+    parser_train.add_argument('--poisson_weight', required=True, type=float,
+                              help='relative weight of poisson loss')
+    # validation args
+    parser_train.add('--val_files', required=True, type=str,
+                     help='list of data files in the form of [file1, file2, '
+                          '...];'
+                          'or a single path to a folder of files')
+    parser_train.add('--eval_freq', required=True, type=int,
+                     help="evaluation frequency")
+    parser_train.add('--threshold', required=True, type=float,
+                     help="probability threshold above which to call peaks. \
+               Used for classification metrics")
+    parser_train.add_argument('--best_metric_choice', required=True,
+                              type=str,
+                              choices=['BCE', 'MSE', 'Recall',
+                                       'Specificity', 'CorrCoef', 'AUROC'],
+                              help="metric to select the best model.\
+                                Choice is case sensitive.")
+    # =========================================================================
+    # Inference args
+    infer_config_path = os.path.join(root_dir, 'configs', 'infer_config.yaml')
+    parser_infer = subparsers.add_parser(
+        'infer',
+        config_file_parser_class=configargparse.YAMLConfigFileParser,
+        default_config_files=[infer_config_path])
+    add_common_options(parser_infer)
+    add_inference_options(parser_infer)
+    # =========================================================================
+    # Evaluation args
+    parser_eval = subparsers.add_parser(
+        'eval',
+        config_file_parser_class=configargparse.YAMLConfigFileParser,
+        default_config_files=[infer_config_path])
+
+    add_common_options(parser_eval)
+    add_inference_options(parser_eval)
```

Comment:
can we add an `add_train_options` and `add_eval_options` function as well? and then those functions can call `add_common_options` from inside them?

Reply:
done